src/pypowsybl_jupyter/networkexplorer.py

```diff
@@ -108,16 +106,30 @@ def on_nadslider_changed(d):
         description='Filter',
         disabled=False,
         continuous_update=True,
-        layout=widgets.Layout(width='350px')
+        layout=widgets.Layout(flex='2%', height='100%', width='350px', margin='1px 0 0 0')
     )
 
+    def update_select_widget(widget, el, elements=None, on_select=None):
+        if on_select:
+            widget.unobserve(on_select, names='value')
+        try:
+            if elements is not None:
+                widget.value = None
```

Comment:
I don't understand why that's needed?

Reply:
this method can  set a widget list's selected value and available values. This  code switches off  the list's events mechanism while those changes are performed, to avoid unwanted triggering of  events.